consensus/utils.go

```diff
@@ -20,19 +20,53 @@ func IsValidSuperMajorityQuorumCertificate(qc QuorumCertificate, validators []Va
 	// Compute the signature that validators in the QC would have signed
 	signaturePayload := GetVoteSignaturePayload(qc.GetView(), qc.GetBlockHash())
 
+	isValidSuperMajority, validatorPublicKeysInQC := checkSuperMajorityStakeForAggregatedSignature(qc.GetAggregatedSignature(), validators)
+	if !isValidSuperMajority {
+		return false
+	}
+
+	return isValidSignatureManyPublicKeys(validatorPublicKeysInQC, qc.GetAggregatedSignature().GetSignature(), signaturePayload[:])
+}
+
+func IsValidSuperMajorityAggregateQuorumCertificate(aggQC AggregateQuorumCertificate, validators []Validator) bool {
+	if !isProperlyFormedAggregateQC(aggQC) || !isProperlyFormedValidatorSet(validators) {
+		return false
+	}
+
+	isValidSuperMajority, signerPublicKeys := checkSuperMajorityStakeForAggregatedSignature(aggQC.GetAggregatedSignature(), validators)
+	if !isValidSuperMajority {
+		return false
+	}
+
+	signedPayloads := [][]byte{}
+	for _, highQCView := range aggQC.GetHighQCViews() {
+		payload := GetTimeoutSignaturePayload(aggQC.GetView(), highQCView)
+		signedPayloads = append(signedPayloads, payload[:])
+	}
+
+	// Validate the signers' aggregate signatures
+	isValidSignature, err := bls.VerifyAggregateSignatureMultiplePayloads(
+		signerPublicKeys,
+		aggQC.GetAggregatedSignature().GetSignature(),
+		signedPayloads,
+	)
+	if err != nil || !isValidSignature {
+		return false
+	}
+	return true
+}
+
+func checkSuperMajorityStakeForAggregatedSignature(aggSig AggregatedSignature, validators []Validator) (bool, []*bls.PublicKey) {
```

Comment:
Simplify this to just pass in the signers list and validator set. This way it's clearer from the function signature alone exactly what the function does
```suggestion
func isSuperMajorityStakeSignersList(signersList *bitset.Bitset, validators []Validator) (bool, []*bls.PublicKey) {
```